src/components/checkbox-group/checkbox-group.test.ts

```diff
@@ -0,0 +1,298 @@
+      </form>
+    `);
+    const button = form.querySelector('sl-button')!;
+    const checkbox = form.querySelectorAll('sl-checkbox')[1]!;
+    const submitHandler = sinon.spy((event: SubmitEvent) => {
+      formData = new FormData(form);
+      event.preventDefault();
+    });
+    let formData: FormData;
+
+    form.addEventListener('submit', submitHandler);
+    checkbox.click();
+    button.click();
+    await waitUntil(() => submitHandler.calledOnce);
+
+    expect(formData!.getAll('a')).to.include('1: true');
+    expect(formData!.getAll('a')).to.include('2: true');
+  });
+
+  it('should be present in form data when using the form attribute and located outside of a <form>', async () => {
+    const el = await fixture<HTMLFormElement>(html`
+      <div>
+        <form id="f">
+          <sl-button type="submit">Submit</sl-button>
+        </form>
+        <sl-checkbox-group form="f" name="a">
+          <sl-checkbox id="checkbox-1" value="1" checked></sl-checkbox>
+          <sl-checkbox id="checkbox-2" value="2" checked></sl-checkbox>
+          <sl-checkbox id="checkbox-3" value="3"></sl-checkbox>
+        </sl-checkbox-group>
+      </div>
+    `);
+    const form = el.querySelector('form')!;
+    const formData = new FormData(form);
+
+    expect(formData.getAll('a')).to.include('1: true');
+    expect(formData.getAll('a')).to.include('2: true');
+  });
+});
+
+describe('when the value changes', () => {
+  it('should emit sl-change when toggled with the space bar', async () => {
+    const checkboxGroup = await fixture<SlCheckboxGroup>(html`
+      <sl-checkbox-group>
+        <sl-checkbox id="checkbox-1" value="1"></sl-checkbox>
+        <sl-checkbox id="checkbox-2" value="2"></sl-checkbox>
+      </sl-checkbox-group>
+    `);
+    const firstCheckbox = checkboxGroup.querySelector<SlCheckbox>('#checkbox-1')!;
+    const changeHandler = sinon.spy();
+    const inputHandler = sinon.spy();
+
+    checkboxGroup.addEventListener('sl-change', changeHandler);
+    checkboxGroup.addEventListener('sl-input', inputHandler);
+    firstCheckbox.focus();
+    await sendKeys({ press: ' ' });
+    checkboxGroup.dispatchEvent(new FocusEvent('blur'));
+    await checkboxGroup.updateComplete;
+
+    expect(changeHandler).to.have.been.called;
+    expect(inputHandler).to.have.been.called;
+    expect(checkboxGroup.value).to.include('1: true');
+    expect(checkboxGroup.value).to.include('2: false');
+  });
+
+  it('should emit sl-change and sl-input when clicked', async () => {
+    const checkboxGroup = await fixture<SlCheckboxGroup>(html`
+      <sl-checkbox-group>
+        <sl-checkbox id="checkbox-1" value="1"></sl-checkbox>
+        <sl-checkbox id="checkbox-2" value="2"></sl-checkbox>
+      </sl-checkbox-group>
+    `);
+    const checkbox = checkboxGroup.querySelector<SlCheckbox>('#checkbox-1')!;
+    setTimeout(() => checkbox.click());
+    const event = (await oneEvent(checkboxGroup, 'sl-change')) as SlChangeEvent;
```

Comment:
I'm missing where we're testing `sl-input` in this one

Reply:
This is another one where I was following the pattern in the same test for [src/components/radio-group/radio-group.test.ts](src/components/radio-group/radio-group.test.ts), but maybe it's a mistake in the Radio Group test? I just updated both to include an event listener for `sl-input`. Pushed a new commit!